Softeq.XToolkit.WhiteLabel.Tests/Helpers/AvatarPlaceholderBuilderTests/AvatarPlaceholderBuilderTests.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using Softeq.XToolkit.WhiteLabel.Helpers;
+using Softeq.XToolkit.WhiteLabel.Tests.TestUtils;
+using Xunit;
+
+namespace Softeq.XToolkit.WhiteLabel.Tests.Helpers.AvatarPlaceholderBuilderTests
+{
+    public class AvatarPlaceholderBuilderTests
+    {
+        #region GetAbbreviation
+
+        [Fact]
+        public void GetAbbreviation_WithNullName_ThrowsCorrectException()
+        {
+            Assert.Throws<ArgumentNullException>(()
+                => AvatarPlaceholderBuilder.GetAbbreviation(null));
+        }
+
+        [Theory]
+        [MemberData(
+            nameof(AvatarPlaceholderBuilderDataProvider.EmptyOrWhiteSpaceNameData),
+            MemberType = typeof(AvatarPlaceholderBuilderDataProvider))]
+        public void GetAbbreviation_WithEmptyOrWhiteSpaceName_ReturnsEmptyAbbreviation(string name)
+        {
+            var abbr = AvatarPlaceholderBuilder.GetAbbreviation(name);
+
+            Assert.NotNull(abbr);
+            Assert.Empty(abbr);
+        }
+
+        [Theory]
+        [MemberData(
+            nameof(AvatarPlaceholderBuilderDataProvider.NonEmptyNameData),
+            MemberType = typeof(AvatarPlaceholderBuilderDataProvider))]
+        public void GetAbbreviation_WithNonEmptyName_ReturnsValidAbbreviation(string name)
+        {
```

Comment:
Please use `input`, `expected` name params to avoid `foreach`, and a lot of asserts.

Reply:
I don't really know what is the expected value for GetAbbreviation method. I can make a guess, but I can't be sure that the logic, that is used in this method is the same that I use. That's why I'm only checking the things that I know for sure:

- Abbreviation length
- Only uppercase letters
- Abbreviation should not contain any characters that doesn't exist in the input string

My proposal is to replace foreach with `Assert.All` checks:

```
Assert.All(abbr, (c) => char.IsLetterOrDigit(c));
Assert.All(abbr, (c) => char.IsUpper(c));
Assert.All(abbr, (c) => name.Contains(c, StringComparison.InvariantCultureIgnoreCase));
```

What do you think?

Reply:
I mean if we have input value "First Last" we expect that abbreviation will be "FL". In this case test data can provide input and expected value without additional checks.